src/Microsoft.AspNet.Hosting/WebHostConfiguration.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.AspNet.Hosting
+{
+    public class WebHostConfiguration
```

Comment:
You could just have a string property that maps to the key in config, and then expose the current logic just like you do now.

```
public string DetailedErrors { get; set; }

public bool IsDetailed { get { <same logic as you have> }
```